packages/node_modules/@ciscospark/plugin-messages/test/integration/spec/messages.js

```diff
@@ -215,5 +217,86 @@ describe('plugin-messages', function () {
           assert.instanceOf(reason, SparkHttpError.NotFound);
         }));
     });
+
+    describe('events', () => {
+      describe('#listen()', () => {
+        [
+          'post',
+          'share'
+        ].forEach((verb) => {
+          it(`emits a "created" event when a "${verb}" activity arrives`, () => {
+            const event = makeActivityEvent(verb);
+            const spy = sinon.spy();
+            spark.messages.listen()
+              .then(() => {
+                spark.messages.on('created', spy);
+                debug(`create mock activity event: ${JSON.stringify(event)}`);
+                spark.internal.mercury._onmessage(event);
+
+                return Promise.resolve()
+                  .then(() => {
+                    assert.calledOnce(spy);
+                  });
+              });
+          });
+        });
+      });
+
+      describe('#createMessagesEventData', () => {
+        [
+          {
+            tags: ['ONE_ON_ONE'],
+            roomType: 'direct'
+          },
+          {
+            tags: [],
+            roomType: 'group'
+          }
+        ].forEach((input) => {
+          it(`creates a messages event for a ${input.roomType} room`, () => {
+            const activityEvent = makeActivityEvent('post', input.tags);
+            const {activity} = activityEvent.data.data;
+            const event = spark.messages.createMessagesEventData(activity);
+
+            debug(`messages event created: ${JSON.stringify(event)}`);
```

Comment:
This needed?

Reply:
It only prints to screen if you use `DEBUG=messages` from the CLI. I found it helpful.